./runall interpreter.sml
==== tests/1_basics/eval01.df ====
==== tests/1_basics/eval02.df ====
==== tests/1_basics/eval03.df ====
==== tests/1_basics/eval04.df ====
==== tests/1_basics/eval05.df ====
==== tests/1_basics/eval06.df ====
--- tests/1_basics/eval06.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/1_basics/eval06.output	2010-02-26 10:03:29.000000000 -0800
@@ -1,2 +1,2 @@
 9
-100
+9
==== tests/1_basics/eval07.df ====
==== tests/1_basics/eval08.df ====
--- tests/1_basics/eval08.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/1_basics/eval08.output	2010-02-26 10:03:30.000000000 -0800
@@ -1 +1 @@
-unit
+<invalid>
==== tests/1_basics/eval09.df ====
==== tests/2_advanced/eval01.df ====
--- tests/2_advanced/eval01.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/2_advanced/eval01.output	2010-02-26 10:03:32.000000000 -0800
@@ -1 +1 @@
-120
+operator '*' requires int * int, found int * unit
==== tests/2_advanced/eval02.df ====
--- tests/2_advanced/eval02.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/2_advanced/eval02.output	2010-02-26 10:03:33.000000000 -0800
@@ -1,4 +1,4 @@
 2
 3
 7
-7
+3
==== tests/2_advanced/eval03.df ====
==== tests/2_advanced/eval04.df ====
--- tests/2_advanced/eval04.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/2_advanced/eval04.output	2010-02-26 10:03:35.000000000 -0800
@@ -2,4 +2,4 @@
 2
 8
 2
-8
+3
==== tests/2_advanced/eval05.df ====
--- tests/2_advanced/eval05.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/2_advanced/eval05.output	2010-02-26 10:03:36.000000000 -0800
@@ -19,5 +19,233 @@
 6
 4
 2
+20
+19
+38
+36
+34
+32
+30
+28
+26
+24
+22
+20
+18
+16
+14
+12
+10
+8
+6
+4
+2
+19
+18
+36
+34
+32
+30
+28
+26
+24
+22
+20
+18
+16
+14
+12
+10
+8
+6
+4
+2
+18
+17
+34
+32
+30
+28
+26
+24
+22
+20
+18
+16
+14
+12
+10
+8
+6
+4
+2
+17
+16
+32
+30
+28
+26
+24
+22
+20
+18
+16
+14
+12
+10
+8
+6
+4
+2
+16
+15
+30
+28
+26
+24
+22
+20
+18
+16
+14
+12
+10
+8
+6
+4
+2
+15
+14
+28
+26
+24
+22
+20
+18
+16
+14
+12
+10
+8
+6
+4
+2
+14
+13
+26
+24
+22
+20
+18
+16
+14
+12
+10
+8
+6
+4
+2
+13
+12
+24
+22
+20
+18
+16
+14
+12
+10
+8
+6
+4
+2
+12
+11
+22
+20
+18
+16
+14
+12
+10
+8
+6
+4
+2
+11
+10
+20
+18
+16
+14
+12
+10
+8
+6
+4
+2
+10
+9
+18
+16
+14
+12
+10
+8
+6
+4
+2
+9
+8
+16
+14
+12
+10
+8
+6
+4
+2
+8
+7
+14
+12
+10
+8
+6
+4
+2
+7
+6
+12
+10
+8
+6
+4
+2
+6
+5
+10
+8
+6
+4
+2
+5
+4
+8
+6
+4
+2
+4
+3
+6
+4
+2
+3
+2
+4
+2
+2
+1
+2
+1
 0
--1
==== tests/2_advanced/eval06.df ====
--- tests/2_advanced/eval06.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/2_advanced/eval06.output	2010-02-26 10:03:36.000000000 -0800
@@ -1,14 +1,8 @@
+attempt to invoke variable 'j' as a function
 999
-3
-999
-3
 999
-3
 999
 999
-17
-999
-17
 999
 17
 999
==== tests/2_advanced/eval07.df ====
==== tests/2_advanced/eval08.df ====
--- tests/2_advanced/eval08.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/2_advanced/eval08.output	2010-02-26 10:03:38.000000000 -0800
@@ -1,4 +1,4 @@
 2
-3
+2
 2
 3
==== tests/3_return/eval01.df ====
--- tests/3_return/eval01.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/3_return/eval01.output	2010-02-26 10:03:39.000000000 -0800
@@ -1 +1,7 @@
-120
+return not supported at this time
+return not supported at this time
+return not supported at this time
+return not supported at this time
+return not supported at this time
+return not supported at this time
+<invalid>
==== tests/3_return/eval02.df ====
--- tests/3_return/eval02.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/3_return/eval02.output	2010-02-26 10:03:40.000000000 -0800
@@ -1 +1,2 @@
-120
+return not supported at this time
+<invalid>
==== tests/3_return/eval03.df ====
--- tests/3_return/eval03.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/3_return/eval03.output	2010-02-26 10:03:41.000000000 -0800
@@ -1,3 +1,7 @@
-3
-7
-10
+return not supported at this time
+return not supported at this time
+return not supported at this time
+return not supported at this time
+<invalid>
+<invalid>
+<invalid>
==== tests/3_return/eval04.df ====
--- tests/3_return/eval04.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/3_return/eval04.output	2010-02-26 10:03:42.000000000 -0800
@@ -1,7 +1,11 @@
+return not supported at this time
+return not supported at this time
+return not supported at this time
+return not supported at this time
+0
+<invalid>
+0
+<invalid>
+0
+<invalid>
 0
-3
-1
-5
-2
-9
-4
==== tests/3_return/eval05.df ====
--- tests/3_return/eval05.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/3_return/eval05.output	2010-02-26 10:03:43.000000000 -0800
@@ -1,9 +1,23 @@
+return not supported at this time
+return not supported at this time
+return not supported at this time
+return not supported at this time
+return not supported at this time
+return not supported at this time
+return not supported at this time
+return not supported at this time
 3
-unit
+2
+1
+<invalid>
 10
 9
 8
 7
 6
 5
-unit
+4
+3
+2
+1
+<invalid>
==== tests/3_return/eval06.df ====
--- tests/3_return/eval06.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/3_return/eval06.output	2010-02-26 10:03:44.000000000 -0800
@@ -1,2 +1,4 @@
+return not supported at this time
 1
-2
+3
+<invalid>
==== tests/3_return/eval07.df ====
--- tests/3_return/eval07.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/3_return/eval07.output	2010-02-26 10:03:44.000000000 -0800
@@ -1 +1,2 @@
-3
+return not supported at this time
+<invalid>
==== tests/3_return/eval08.df ====
--- tests/3_return/eval08.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/3_return/eval08.output	2010-02-26 10:03:45.000000000 -0800
@@ -1,2 +1,2 @@
-3
-8
+return not supported at this time
+operator '+' requires int * int, found <invalid> * int
==== tests/3_return/eval09.df ====
--- tests/3_return/eval09.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/3_return/eval09.output	2010-02-26 10:03:46.000000000 -0800
@@ -1,2 +1,2 @@
-3
-8
+return not supported at this time
+operator '+' requires int * int, found <invalid> * int
==== tests/4_dynamic_typing/badsemantic01.df ====
==== tests/4_dynamic_typing/badsemantic02.df ====
==== tests/4_dynamic_typing/badsemantic03.df ====
==== tests/4_dynamic_typing/badsemantic04.df ====
==== tests/4_dynamic_typing/badsemantic05.df ====
==== tests/4_dynamic_typing/badsemantic06.df ====
==== tests/4_dynamic_typing/badsemantic07.df ====
==== tests/4_dynamic_typing/badsemantic08.df ====
==== tests/4_dynamic_typing/badsemantic09.df ====
--- tests/4_dynamic_typing/badsemantic09.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/4_dynamic_typing/badsemantic09.output	2010-02-26 10:03:54.000000000 -0800
@@ -1 +1 @@
-operator '+' requires int * int, found int * unit
+operator '+' requires int * int, found int * <invalid>
==== tests/4_dynamic_typing/badsemantic10.df ====
--- tests/4_dynamic_typing/badsemantic10.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/4_dynamic_typing/badsemantic10.output	2010-02-26 10:03:55.000000000 -0800
@@ -1 +1,2 @@
-operator '+' requires int * int, found bool * int
+return not supported at this time
+operator '+' requires int * int, found <invalid> * int
==== tests/4_dynamic_typing/badsemantic11.df ====
--- tests/4_dynamic_typing/badsemantic11.correct	2010-02-25 13:18:44.000000000 -0800
+++ tests/4_dynamic_typing/badsemantic11.output	2010-02-26 10:03:56.000000000 -0800
@@ -1 +1,2 @@
-operator '+' requires int * int, found int * bool
+return not supported at this time
+<invalid>
==== tests/4_dynamic_typing/badsemantic12.df ====
